src/services/users/generate-password-reset-link.js

```diff
@@ -0,0 +1,35 @@
+const db = require("../../db")
+const config = require("../../config")
+const { AppError } = require("../../lib")
+const generateAccessToken = require("./generate-access-token")
+
+/**
+ * Generates a new password reset link
+ * @param {string} email - email of the user
+ * @returns {string} - password reset link
+ */
+
+
+const generateResetLink = async (email) => {
+
+	const { rows } = await db.query(
+		"SELECT * FROM users WHERE email = $1",
+		[email]
+	)
+
+	const userProfile = rows[0]
+
+	if (!userProfile) {
+		throw new AppError('Invalid email', 401)
```

Comment:
so the service shouldn't know about the status codes, that should happen in the routes

that's because we'll not necessarily be using services just for API routes, we could have background jobs and other workers which could call our services. And those ones have nothing to do with a status code 

that's why we throw custom errors and translate them into status codes in the routes

Reply:
also if we want to separate this, it should be related to the specific function

as in `send-password-reset-link` will now be a folder, with an `index.js` , and then the `generate-password-reset-link.js` will also be a file in that folder

that will be the case till we're sure `generate-password-reset-link` is being shared between one or more functions